### ISSUE TITLE
Conversion from UTF8 to ASCII drops some symbols

### ISSUE THREAD

**Issue by tobadia** [nutterb/redcapAPI#99]
Hey again @nutterb !

Back from holiday and long overdue on looking at all the edits you made. Upon looking at issue #92 that you addressed, I stumbled upon other things that seem to have been introduced when you refactored the `exportRecords()` method, especially when passing calls to `iconv()` to convert the content returned by the API call from UTF-8 to ASCII: some symbols are silently dropped, e.g. the "°" degree symbol.

At first I thought it was `utils::read.csv()` and the `colClasses` approach but the `iconv()` call seems to be the culprit. Why go to ASCII anyway? UTF-8 seems like a decent option, all the more when it supports fancy characters that are not available in the ASCII space (I'm thinking about other alphabets like Khmer (yes, I'm working on a Cambodian study :-)). Simply dropping the iconv call seems to be OK and does not seem to lead to other issues. What was the rationale behind moving to ASCII ?

In my case, the issue appeared on manual GPS coordinates inputted (not using the @LONGITUTDE and @LATITUDE action tags from REDCap): inputs like "12°12.120" get recasted to "1212.120", obviously leading to erroneous content afterwards.

Best
Thomas

**Comment by nutterb** [comment]
Hmmm...I can see why that is problematic.  

The motivation was based on what appears to be problems with reading some projects. I have a number of bug reports where `exportRecords` fails, but when I get ahold of the meta data and build a project to explore, it works just fine.  As far as I can tell, it is related to weird characters being introduced through the REDCap API.  I think one notable culprit is when people copy bullet points out of MS word into the UI.  `iconv` was an attempt to bludgeon the data into submission.  

You have a point, however, that perhaps that should only be applied to the meta data, and leave the actual data alone.

**Comment by tobadia** [comment]
Indeed, tackling only meta-data with the iconv shotgun would be an option. I feel for you, having to cope with copy/paste from word documents... That's just wrong (╯°□°）╯︵ ┻━┻

**Comment by nutterb** [comment]
Give `devtools::install_github("nutterb/redcapAPI", ref = "devel")` a try.  Should be version 2.0.1

**Comment by tobadia** [comment]
Works like a charm indeed. Also somehow linking back to issue #92 (that I just involuntarily re-opened by commenting seconds ago), I still have some non-latin characters that are dropped silently from value labels in checkboxes. Could this also be due to this `iconv` thing ? I'm guessing the values for checkboxes could very well be extracted from the metadata, that still get iconv'ed.

This is a minor issue though. Maybe even not an issue if you choose to impose the iconv on metadata, if that's a better option than not having it and risking corrupted metadata.

**Comment by nutterb** [comment]
As I've thought more about it, I think forcing everything to ASCII is probably overly harsh.  I haven't actually confirmed that UTF-8 characters are a problem. It's probably something beyond that set that is the issue.  I will look at ways to drop anything that isn't in either UTF-8 or ASCII character sets, but I can probably find a way to retain those.

**Comment by tobadia** [comment]
Happy to try out anything that you'd implement. I have a huge project with entries in Khmer so that's a good way to toy around for that specific matter.
That was also the occasion to discover that R/Rstudio on a mac handles Khmer symbols much better than could have been expected.

**Comment by tobadia** [comment]
Hey,

So I've seen your update one issue #104 and indeed what you had in place before made sense. And the edit you've implemented now that the "autocomplete" feature has been propagated in the metadata also seems to work (I can't notice anything broken in my data so far and my quality routines work as they did before).

Going back to the ASCII/UTF8 issue though, even passing it through the metadata seems a bit overkill. There are a bunch of characters that don't translate into ASCII (fractions for instance), and when a list/radio button/checkbox has coded values including such characters, they are dropped, sometimes resulting in empty strings, then colliding as "duplicate factor levels".

I'm not exactly sure what approach would be best. Obviously you've made me realise using such chars in my project as level isn't the right choice, however the UTF8 palette is still somewhat versatile enough to be assumed safe (at least that's how I feel). Did you ever manage to replicate the bugs you mentioned above with stuffs like bullets etc copy/pasted from external files?

**Comment by nutterb** [comment]
I've not succeeded in replicating it, but I have another issue open where i couldnt even get the meta data out of REDCap until I manually downloaded the csv, scrubbed it with `iconv`, then reuploaded it. Talk about overkill.

I'm going to take the easy way out and declare that there appears to be a neutral zone between what REDCap pushes out and what R can handle gracefully, and I'm not motivated enough to tame that wilderness.

For now, I've backpedaled some and the devel branch makes utf scrubbing optional with an argument. That should make it possible to clean up inadvertent messes from the copy/paste crowd without stepping on the toes of the more technically savvy
